File: Runtime/TransitionSpatialInitiationMultiplierCollection.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'************************************************************************************

Imports System.Collections.ObjectModel

Friend Class TransitionSpatialInitiationMultiplierCollection
    Inherits KeyedCollection(Of Integer, TransitionSpatialInitiationMultiplier)


    Protected Overrides Function GetKeyForItem(ByVal item As TransitionSpatialInitiationMultiplier) As Integer
        Return item.TransitionSpatialInitiationMultiplierId
    End Function

End Class

File: Runtime/TransitionSpatialInitiationMultiplierCollection.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System.Collections.ObjectModel;

namespace SyncroSim.STSim
{
    internal class TransitionSpatialInitiationMultiplierCollection : KeyedCollection<int, TransitionSpatialInitiationMultiplier>
    {
        protected override int GetKeyForItem(TransitionSpatialInitiationMultiplier item)
        {
            return item.TransitionSpatialInitiationMultiplierId;
        }
    }
}
